imap_processing/lo/l0/lo_science.py

```diff
@@ -148,3 +153,167 @@ def decompress(
     )
 
     return decompressed_ints
+
+
+def parse_events(dataset: xr.Dataset, attr_mgr: ImapCdfAttributes) -> xr.Dataset:
+    """
+    Parse and decompress binary direct event data for Lo.
+
+    Parameters
+    ----------
+    dataset : xr.Dataset
+        Lo science direct events from packets_to_dataset function.
+    attr_mgr : ImapCdfAttributes
+        CDF attribute manager for Lo L1A.
+
+    Returns
+    -------
+    dataset : xr.Dataset
+        Parsed and decompressed direct event data.
+    """
+    # TODO: Add logging. Want to wait until I have a better understanding of how the
+    #  DEs spread across multiple packets will work first
+
+    # Sum each count to get the total number of direct events for the pointing
+    num_de: int = np.sum(dataset["count"].values)
+
+    de_fields = [
+        "coincidence_type",
+        "de_time",
+        "esa_step",
+        "mode",
+        "tof0",
+        "tof1",
+        "tof2",
+        "tof3",
+        "cksm",
+        "pos",
+    ]
+    # Initialize all Direct Event fields with their fill value
+    # L1A Direct event data will not be tied to an epoch
+    # data will use a direct event index for the pointing as its coordinate/dimension
+    for field in de_fields:
+        dataset[field] = xr.DataArray(
+            np.full(num_de, attr_mgr.get_variable_attributes(field)["FILLVAL"]),
+            dims="direct_events",
+        )
+
+    # The DE index for the entire pointing
+    pointing_de = 0
+    # for each direct event packet in the pointing
+    for pkt_idx, de_count in enumerate(dataset["count"].values):
```

Comment:
I would prefer for this section to be in a different method - mainly for documentation. Overall it's pretty clear what's happening here, though! Is it possible to loop through the values of `DATA_BITS` rather than writing them all out here? You have a lot of `dataset["{val}"] = parse_de_bin(dataset, pkt_idx, bit_pos, {val}` that could reduce duplication. The last set of if statements can probably work the same way.

Reply:
If you do greatly simplify this for loop then it's fine in this method

Reply:
Agreed. I'll clean this up